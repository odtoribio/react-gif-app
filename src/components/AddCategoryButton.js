import React, { useState } from 'react'
import PropTypes from 'prop-types'  

export const AddCategoryButton = ({ addCategory }) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputValue = (e) => {
    setInputValue(e.target.value)
  }

  const handleOnSubmit = (e) => {
    e.preventDefault()

    if (inputValue){
      addCategory(inputValue)
      setInputValue('')
    }
  }
  
  return (
    <form onSubmit={ handleOnSubmit }>
      <input 
        type="text"
        value={ inputValue }
        onChange= { handleInputValue }
      />
    </form>
  )
}

AddCategoryButton.propTypes = {
  addCategory: PropTypes.func.isRequired,
}
