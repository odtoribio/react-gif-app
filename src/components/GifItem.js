import React from 'react'
import PropTypes from 'prop-types';

export const GifItem = (props) => {
  const { title, url } = props;
  return (
    <div className="card animate__animated animate__zoomIn">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}


GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
