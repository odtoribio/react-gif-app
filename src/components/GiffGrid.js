import React from 'react';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GiffGrid = ( { category } ) => {

  const { data:gifs, loading } = useFetchGifs(category)

  return (
    <>
      <h3>{ category }</h3>
      { loading && 'Loading...' }
      <div className="card-grid">
          {
            gifs.map( gif => (
              <GifItem 
                key={gif.id} 
                gif={ gif }
              />
            ))
          }
      </div>
    </>
  )
}
