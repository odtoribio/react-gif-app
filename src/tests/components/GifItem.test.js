import React from 'react'
import {shallow} from 'enzyme'
import {GifItem} from '../../components/GifItem'
describe("Test Gif Item component",() => {
  
  const gif = {url:'http://theurl.com', title:'title'}
  const GifItemComponent = shallow(<GifItem title={gif.title} url={gif.url}/>)
  test('should render GifItem component', () => {

    expect(GifItemComponent).toMatchSnapshot()
  })

  test('should has a paragraph with title', () => {
    const paragraph = GifItemComponent.find('p')
    expect(paragraph.text().trim()).toBe(gif.title);
  })
  
  test('should have image url and image alt text', () => {
    const img = GifItemComponent.find('img');

    expect(img.prop('src')).toBe(gif.url);
    expect(img.prop('alt')).toBe(gif.title);
  })
  
  test('should have classname animate__zoomIn', () => {
    const div = GifItemComponent.find('div');

    const className = div.prop('className')
    expect(className.includes('animate__zoomIn')).toBe(true);
  })
  
})