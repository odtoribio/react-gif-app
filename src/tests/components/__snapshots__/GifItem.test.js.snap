// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Gif Item component should render GifItem component 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="title"
    src="http://theurl.com"
  />
  <p>
    title
  </p>
</div>
`;
